conans/model/requires.py

```diff
@@ -276,20 +276,7 @@ def transform_downstream(self, pkg_type, require, dep_pkg_type):
             return
 
         # Regular and test requires
-        if dep_pkg_type is PackageType.SHARED:
-            if pkg_type is PackageType.SHARED:
```

Comment:
The implementation was duplicated for both SHARED and STATIC, as the traits account for it already.